### PR TITLE
Add pure metrics computation and /analyze endpoint

### PR DESCRIPTION
**Pull request by slate71**
## Summary
- Implements pure metrics computation for 48h PR metrics from GitHub events
- Adds POST /analyze endpoint that queries recent events and returns metrics + events
- Comprehensive unit testing with pytest coverage
- Response time well under 200ms requirement (~12ms measured)

## Changes
- **api/metrics.py**: Core metrics computation with `compute_48h_metrics()` function
- **api/main.py**: Updated POST /analyze endpoint with database integration  
- **api/tests/test_metrics.py**: 7 comprehensive unit tests covering edge cases
- **api/requirements.txt**: Added pytest for testing

## Features
- Computes PR opened/merged counts from last 48h of GitHub events
- Handles multiple timestamp formats (Z and +00:00)  
- Filters and sorts events by recency
- Stub values for review hours and Linear metrics (ready for future implementation)
- Robust error handling and edge case coverage

## Test Results
```
7 tests passed, response time ~12ms, metrics verified against manual SQL queries
```

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: api/metrics.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                dt = datetime.fromisoformat(ts_str)
+                if dt.tzinfo is None:
+                    dt = dt.replace(tzinfo=timezone.utc)
+                return dt
+        except (ValueError, TypeError):
+            # Return epoch for invalid timestamps (will be sorted last)
+            return datetime.fromtimestamp(0, tz=timezone.utc)
+    
+    sorted_events = sorted(events, key=parse_timestamp, reverse=True)
+    return sorted_events[:limit]
```
Comment (slate71) [inline api/metrics.py]: New line.

Comment (slate71) [reply api/metrics.py]: Make sure all files have new lines.
